compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -1443,7 +1442,7 @@ fn suggest_ampmut<'tcx>(
     decl_span: Span,
     opt_assignment_rhs_span: Option<Span>,
     opt_ty_info: Option<Span>,
-) -> (bool, Span, String) {
+) -> Option<(bool, Span, String, Option<(Span, String)>)> {
```

Comment:
Suggestion: usually I don't mind as much for suggestion code, but IMO this type is complicated enough to warrant pulling it out to its own struct. When I read the constructions below, I am left head scratching as to what the `true`s and `None`s or `span`s actually refer to.

Maybe something like

```rs
struct AmpMutSugg {
    binding_exists: bool, /* i didn't look too closely at what this bool means */
    span: Span, /* prob a more meaningful name */
    sugg: String,
    extra_label: Option<(Span, String)>, /* im not very sure what this one does */
}
```

Reply:
I actually contemplated doing that initially but end up not doing it since I wanted the diff to be minimal. Now that you ask for it I'm more than happy to add it. Done!